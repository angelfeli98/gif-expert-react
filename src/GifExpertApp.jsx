
import { useState } from 'react';

import GridGif from './components/GridGif';
import CategoryInput from './components/CategoryInput';

const GifExpertApp = () => {

    const [categories, setCategories] = useState([]);

    const onNewValue = (value) => {
        if(categories.includes(value))
            return;
        setCategories([value, ...categories]);
    };

    return (
        <>
            <h1>GifExpertApp</h1>
            <CategoryInput onNewValue={onNewValue} />
            <div>
                {
                    categories.map(category =>
                        <GridGif key={category} category={category} />
                    )
                }
            </div>
        </>
    )
}

export default GifExpertApp;
