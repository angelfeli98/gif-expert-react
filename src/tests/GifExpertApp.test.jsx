/* eslint-disable no-undef */

import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import '@testing-library/jest-dom';

import GifExpertApp from "../GifExpertApp";

jest.mock('../utils/env', () => ({key: 'qPsZQJFTytqz0J6TqTmCZb58dTWaqyRX', baseUrl: 'https://api.giphy.com/v1/gifs/search'}));

describe('<GifExpertApp/>', () => {
    test('should render', () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    })

    test('should complete flow', async() => {
        render(<GifExpertApp />);
        fireEvent.change(screen.getByRole('textbox'), {target: {value: 'Morty'}});
        fireEvent.keyUp(screen.getByRole('textbox'), {key: 'Enter'});
        await waitFor(() => {
            expect(screen.getByRole('heading', {level: 2})).toHaveTextContent('Morty');
        })
    })

    test('should not complete flow', async() => {
        render(<GifExpertApp />);
        fireEvent.change(screen.getByRole('textbox'), {target: {value: ''}});
        fireEvent.keyUp(screen.getByRole('textbox'), {key: 'Enter'});
        await waitFor(() => {
            expect(screen.queryByText('Morty')).toBeNull();
        })
    })
})