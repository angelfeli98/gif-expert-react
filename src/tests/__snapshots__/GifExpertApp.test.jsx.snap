// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp/> should render 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <input
    placeholder="Categoria"
    type="text"
    value=""
  />
  <div />
</div>
`;
